src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -83,13 +82,17 @@ IWorkspaceProjectContext IWorkspaceProjectContextFactory.CreateProjectContext(st
 
             if (languageName == LanguageNames.FSharp)
             {
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
                 var shell = await _serviceProvider.GetServiceAsync<SVsShell, IVsShell7>().ConfigureAwait(true);
 
                 // Force the F# package to load; this is necessary because the F# package listens to WorkspaceChanged to 
                 // set up some items, and the F# project system doesn't guarantee that the F# package has been loaded itself
                 // so we're caught in the middle doing this.
                 var packageId = Guids.FSharpPackageId;
                 await shell.LoadPackageAsync(ref packageId);
+
+                await TaskScheduler.Default;
```

Comment:
❗ This is in direct conflict with the following comment

Reply:
Ah good catch. Will investigate.

Reply:
I've removed the need for CPSProject's constructor to be on the UI thread, and removed the comment.

Reply:
this feels like this should move right before we call CPS constructor.  that way we do this for all languages, not just F#.